packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/index.js

```diff
@@ -43,6 +56,35 @@ export default function SidebarNavigationScreenNavigationMenu() {
 
 	const menuTitle = navigationMenu?.title?.rendered || navigationMenu?.slug;
 
+	const handleSave = async ( edits = {} ) => {
+		editEntityRecord( 'postType', postType, postId, edits );
+		await saveEditedEntityRecord( 'postType', postType, postId );
+		createSuccessNotice( __( 'Renamed Navigation menu' ), {
+			type: 'snackbar',
+		} );
+	};
```

Comment:
Let's seperate the concerns of entity handling and post-operationg actions. We can provide a single method which coordinates both actions but they should be split up to aid in maintainability

Reply:
Let's do this in a followup. This PR is already pretty large.